src/lib/operators/table_scan/column_like_table_scan_impl.cpp

```diff
@@ -55,11 +55,14 @@ void ColumnLikeTableScanImpl::_scan_generic_segment(
       using ColumnDataType = typename decltype(iter)::ValueType;
 
       if constexpr (std::is_same_v<ColumnDataType, pmr_string>) {
-        _matcher.resolve(_invert_results, [&](const auto& resolved_matcher) {
-          const auto functor = [&](const auto& position) {
-            return resolved_matcher(position.value());
-          };
-          _scan_with_iterators<true>(functor, iter, end, chunk_id, matches);
+        LikeMatcher::resolve_condition(predicate_condition, [&](const auto& predicate) {
+          using Predicate = std::decay_t<decltype(predicate)>;
+          LikeMatcher::resolve_pattern<Predicate>(_pattern, [&](const auto& matcher) {
```

Comment:
Since regex creation is not for free, I wonder if we really want to create it for every single chunk instead of once.
Is there a reason to do it repeatedly?

Reply:
There are a few pros and cons for this approach and the approach taken before, but I think it's best to discuss them offline